src/app/shared/table/actions/table-create-action.ts

```diff
@@ -24,7 +24,7 @@ export class TableCreateAction implements TableAction {
   }
 
   protected create(component: ComponentType<unknown>, dialog: MatDialog,
-    dialogParams: DialogParams<TableData> = {}, refresh?: () => Observable<void>, size?: PopupSize) {
+    dialogParams: DialogParamsWithAuth<TableData, AuthorizationActions> = {}, refresh?: () => Observable<void>, size?: PopupSize) {
```

Comment:
same remark as in Car but with TableData and AuthorizationActions.

Reply:
Problem is that if I only put the AuthorizationActions we won't be able to pass metadata to the dialog if we needed to...   
With this current implementation If we need auth object we pass it along the metadata, ex User:

{ dialogData: { metadata: this.metadata } as User,  **authorizations: { ...authorizations }** }

To be compliant with your previous remarks, we would need to add an optional parameter to the create() and append the authorisations to the dialogParams like:

dialogConfig.data = {
dialogMode: DialogMode.CREATE,
 ...dialogParams **<- contains dialogData : {metadata: { ... }, ... }**
authorizations: { ...authorizations } **<- contains { canList : true }**
};

The **authorization:** would be important otherwise the dialogConfig.data would contain a blob and we would be forced to do something like this in car-dialog.component.ts

public constructor(dialogRef: MatDialogRef<CarDialogComponent>,
    @Inject(MAT_DIALOG_DATA) matDialogParams: **any**) {
    let dialogParams = matDialogParams as DialogParams<Car>;
    let carAuthorizationActions = matDialogParams as CarAuthorizationActions;
}

Let me know ! :)

Reply:
> { dialogData: { metadata: this.metadata } as User, **authorizations: { ...authorizations }** }

We moved authorization out from the data object (Car, User...) for this purpose, why not metadata also :-)?
We can discuss this tomorrow if you are on site :-).